highfidelity/hifi
# Toggle going to 404 redirect domain

- Fix for MS#[18555](https://highfidelity.manuscript.com/f/cases/18555) and #[18689](https://highfidelity.manuscript.com/f/cases/18689)

This PR defaults the behavior of getting redirected to the 404 domain to be disabled.  Enabling it would be by opening the `Developer > Console` after enabling the developer menu via `Settings > Developer Menu`, and setting the mode through `Settings.setValue("enableInterstitialMode", true);`

## TEST PLAN
- Launch Interface
- Enable developer menu in `Settings > Developer Menu`
- Open `Developer > Console` or `CTRL+ALT+J`
- In console, toggle on interstitial mode by using `Window.interstitialModeEnabled = true`
- Press `CTRL+R`
- Navigate to a domain using `GOTO` app
- While going to domain, interstitial page should pop up
- Navigate to a domain that would trigger a timeout, protocol mismatch, or not authorized error
- Interstitial page should pop up, then land you in the 404 redirect domain
## 
- In console, toggle off interstitial mode by using `Window.interstitialModeEnabled = false`
- Press `CTRL+R` to reload all scripts and reload content
- Navigate to a domain using `GOTO` app
- While going to domain, interstitial page should not come up
- Navigate to a domain that would trigger a timeout, protocol mismatch, or not authorized error
- Interstitial page should not come up, then you should see a pop up dialog with the error you triggered
## 
- Try these a couple of times, also with closing/reopening the interface with differing settings from different sessions

Test case:
https://highfidelity.testrail.net/index.php?/cases/view/27807
